### PR TITLE
Rename Azure AD to Entra ID in user-facing text

### PR DESCRIPTION
**Pull request by jingcheng16**
## Product Description

Microsoft has [renamed Azure AD to Entra ID](https://learn.microsoft.com/en-us/entra/fundamentals/new-name). We should update user-facing text (including all text not stored in the database) that references “Azure AD” to “Entra ID”.

## Technical Summary

Tech Spec: https://docs.google.com/document/d/1LOF1lCl3wTsEQkEEsHtZi3u4N6kvjXaLv-Lu6e9aYjo/edit
Ticket: https://dimagi.atlassian.net/browse/SAAS-15237

### Safety story

Just renaming text

### Migrations

- [x] The migrations in this code can be safely applied first independently of the code

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/sso/migrations/0007_alter_identityprovider_idp_type.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 3.2.24 on 2024-03-05 16:06
```
Comment (biyeun) [inline corehq/apps/sso/migrations/0007_alter_identityprovider_idp_type.py:1]: @jingcheng16 can you cherry pick the commit that actually alters the identity provider and create a new PR from that?

Comment (biyeun) [reply corehq/apps/sso/migrations/0007_alter_identityprovider_idp_type.py:1]: FYI matt has a PR open for another migration also coming after `0006...` cc @mjriley https://github.com/dimagi/commcare-hq/pull/34248#event-12067141878

File: corehq/apps/sso/models.py [hunk 2]
```diff
@@ -12,11 +12,11 @@
 
 
 class IdentityProviderType:
-    AZURE_AD = 'azure_ad'
+    AZURE_AD = 'azure_ad'  # Azure AD was the original name
```
Comment (biyeun) [inline corehq/apps/sso/models.py:15]: I think you can update `IdentityProviderType.AZURE_AD` to `IdentityProviderType.ENTRA_ID`

Comment (biyeun) [reply corehq/apps/sso/models.py:15]: suggested comment:
Microsoft renamed Entra ID to Azure ID after implementing this feature

Comment (jingcheng16) [reply corehq/apps/sso/models.py]: Yes, updated in 3f1ea17

**Review by biyeun — APPROVED** [review]
looks good! thank you

**Pull request merged**